### PR TITLE
add folders, remove module mentions

### PR DESCRIPTION
**Pull request by nataliasitko**
Changes proposed in this pull request:

- remove mentions of the Istio module
- add folders and group documentation

### CODE REVIEW

**Review by barchw — APPROVED** [review]

**Review by strekm — APPROVED** [review]
/meow

File: docs/user/02-operation-guides/troubleshooting/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Troubleshooting - Kyma Istio Operator
+
+The troubleshooting section aims to identify the most common recurring problems the users face when they use Kyma Istio Operator, as well as the most suitable solutions to these problems.
+
+If you can't find a solution, don't hesitate to create a [GitHub](https://github.com/kyma-project/istio/issues) issue or reach out to our [Slack channel](http://slack.kyma-project.io/) to get direct support from the community.
```
Comment (IwonaLanger) [inline docs/user/02-operation-guides/troubleshooting/README.md:5]:
```suggestion
If you can't find a solution, don't hesitate to create a [GitHub issue](https://github.com/kyma-project/istio/issues) or reach out to our [Slack channel](http://slack.kyma-project.io/) to get direct support from the community.
```

File: docs/user/00-overview/00-30-overview-istio-sidecars.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 ## Purpose of Istio sidecars
```
[the remark this answers is not in the document]
Comment (nataliasitko) [reply docs/user/00-overview/00-30-overview-istio-sidecars.md:3]: It should be `Istio sidecars` 😄

File: docs/contributor/01-00-installation.md [hunk 3]
```diff
@@ -15,15 +15,15 @@
 - [Docker](https://www.docker.com)
 - [Kyma CLI](https://kyma-project.io/docs/kyma/latest/04-operation-guides/operations/01-install-kyma-CLI)
```
[the remark this answers is not in the document]
Comment (nataliasitko) [reply docs/contributor/01-00-installation.md]: Yes, it seems like the redirect doesn't work with this link. I updated it

File: docs/user/00-overview/00-20-overview-service-mesh.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+The main principle of Istio Service Mesh is to inject Pods of every service with the Envoy sidecar proxy. Envoy intercepts the communication between the services and regulates it by applying and enforcing the rules you create.
+
+By default, Istio installed by Kyma Istio Operator has [mutual TLS (mTLS)](https://istio.io/docs/concepts/security/#mutual-tls-authentication) disabled. See how to [enable sidecar proxy injection](./01-50-enable-sidecar-injection.md). You can manage mTLS traffic in services or at a Namespace level by creating [DestinationRules](https://istio.io/docs/reference/config/networking/destination-rule/) and [Peer Authentications](https://istio.io/docs/tasks/security/authentication/authn-policy/). If you disable sidecar injection for a service or for a Namespace, you must manage their traffic configuration by creating appropriate DestinationRules and Peer Authentications.
```
[the remark this answers is not in the document]
Comment (nataliasitko) [reply docs/user/00-overview/00-20-overview-service-mesh.md]: Thanks, I fixed it

File: docs/user/00-overview/00-30-overview-istio-sidecars.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (nataliasitko) [reply docs/user/00-overview/00-30-overview-istio-sidecars.md:3]: I fixed this lovely typo :)

File: docs/user/00-overview/00-40-overview-istio-setup.md [hunk 5]
```diff
@@ -1,16 +1,16 @@
 # Default Istio setup
 
-This document provides an overview of the default setup for Istio. The Istio module uses the [Istio library](https://github.com/istio/istio/tree/master/operator) for installing Istio. The module includes components like Istiod (Pilot) and Ingress Gateway, which are enabled by default.
+This document provides an overview of the default setup for Istio. Kyma Istio Operator uses the [Istio library](https://github.com/istio/istio/tree/master/operator) for installing Istio. The module includes components like Istiod (Pilot) and Ingress Gateway, which are enabled by default.
```
[the remark this answers is not in the document]
Comment (nataliasitko) [reply docs/user/00-overview/00-40-overview-istio-setup.md]: I replaced it

File: docs/user/02-operation-guides/troubleshooting/03-50-recovering-from-unintentional-istio-module-removal.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+# Kyma Istio Operator was unintentionally disabled
```
[the remark this answers is not in the document]
Comment (nataliasitko) [reply docs/user/02-operation-guides/troubleshooting/03-50-recovering-from-unintentional-istio-module-removal.md]: I updated the whole tutorial

File: docs/contributor/04-10-technical-design.md [hunk 7]
```diff
@@ -12,9 +12,9 @@ In order to transition to a more modularised architecture, the [IstioOperator re
 the [additional istio-resources](https://github.com/kyma-project/kyma/tree/main/resources/istio-resources), and
 the [certificates](https://github.com/kyma-project/kyma/tree/main/resources/certificates) must be moved to the new modules.
 
-#### Istio Operator resource
+#### Kyma Istio Operator resource
 
-The Istio Operator resource is moved into the new Kyma Istio Operator. It is used to define default values for Istio, which the user can customise by modyfying Istio CR.
+The Kyma Istio Operator resource is moved into the new Kyma Istio Operator. It is used to define default values for Istio, which the user can customise by modyfying Istio CR.
```
[the remark this answers is not in the document]
Comment (nataliasitko) [reply docs/contributor/04-10-technical-design.md:17]: It's the technical design of the module, so I guess we can leave in this doc

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**